public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/index.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import NestedResourceTable from './NestedResourceTable';
+import { Row, RowGroup } from './types';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, useStyles2 } from '@grafana/ui';
+import ResourcePickerData from '../../resourcePicker/resourcePickerData';
+import { produce } from 'immer';
+import { Space } from '../Space';
+
+interface ResourcePickerProps {
+  resourcePickerData: Pick<ResourcePickerData, 'getResourcePickerData' | 'getResourcesForResourceGroup'>;
+  resourceURI: string | undefined;
+  // handleSelectResource: (row: Row, isSelected: boolean) => void;
+
+  onApply: (resourceURI: string | undefined) => void;
+  onCancel: () => void;
+}
+
+const ResourcePicker = ({ resourcePickerData, resourceURI, onApply, onCancel }: ResourcePickerProps) => {
+  const styles = useStyles2(getStyles);
+
+  const [rows, setRows] = useState<RowGroup>({});
+  const [internalSelected, setInternalSelected] = useState<string | undefined>(resourceURI);
+
+  useEffect(() => {
+    setInternalSelected(resourceURI);
+  }, [resourceURI]);
+
+  const handleFetchInitialResources = useCallback(async () => {
+    const initalRows = await resourcePickerData.getResourcePickerData();
+    setRows(initalRows);
+  }, [resourcePickerData]);
+
+  useEffect(() => {
+    handleFetchInitialResources();
+  }, [handleFetchInitialResources]);
+
+  const requestNestedRows = useCallback(
+    async (resourceGroup: Row) => {
+      // if we've already fetched resources for a resource group we don't need to re-fetch them
+      if (resourceGroup.children && Object.keys(resourceGroup.children).length > 0) {
+        return;
+      }
+
+      // fetch and set nested resources for the resourcegroup into the bigger state object
+      const resources = await resourcePickerData.getResourcesForResourceGroup(resourceGroup);
+      setRows(
+        produce(rows, (draftState: RowGroup) => {
+          // TODO: fix wonky types
+          (draftState[resourceGroup.subscriptionId].children as RowGroup)[resourceGroup.name].children = resources;
+        })
+      );
+    },
+    [resourcePickerData, rows]
+  );
+
+  const handleSelectionChanged = useCallback((row: Row, isSelected: boolean) => {
+    isSelected ? setInternalSelected(row.id) : setInternalSelected(undefined);
+  }, []);
+
+  const selectedResource = useMemo(() => {
+    if (internalSelected && Object.keys(rows).length) {
+      const matches = /\/subscriptions\/(?<subscriptionId>.+)\/resourceGroups\/(?<selectedResourceGroupName>.+)\/providers\/(?<cloud>.+)/.exec(
```

Comment:
oh yeah here it is, we could probably dry this out somehow

Reply:
OH i actually broke this out into a seperate function to reuse, but never actually used it - `parseResourceURI()` isnt actually used anywhere lol

Reply:
ahh you know what - this was a part of the changes that I (thought I) lost!